tests/sqlparser_mssql.rs

```diff
@@ -187,6 +188,386 @@ fn parse_mssql_create_procedure() {
+                        span: Span::empty(),
+                    }),
+                    data_type: DataType::Varchar(Some(CharacterLength::IntegerLength {
+                        length: 256,
+                        unit: None
+                    })),
+                    default_expr: None,
+                },
+            ]),
+            return_type: Some(DataType::Int(None)),
+            function_body: Some(CreateFunctionBody::AsBeginEnd(BeginEndStatements {
+                begin_token: AttachedToken(TokenWithSpan::wrap(sqlparser::tokenizer::Token::Word(
+                    sqlparser::tokenizer::Word {
+                        value: "BEGIN".to_string(),
+                        quote_style: None,
+                        keyword: Keyword::BEGIN
+                    }
+                ))),
+                statements: vec![
+                    Statement::Set(Set::SingleAssignment {
+                        scope: None,
+                        hivevar: false,
+                        variable: ObjectName::from(vec!["@foo".into()]),
+                        values: vec![sqlparser::ast::Expr::BinaryOp {
+                            left: Box::new(sqlparser::ast::Expr::Identifier(Ident {
+                                value: "@foo".to_string(),
+                                quote_style: None,
+                                span: Span::empty(),
+                            })),
+                            op: sqlparser::ast::BinaryOperator::Plus,
+                            right: Box::new(Expr::Value(number("1").with_empty_span())),
+                        }],
+                    }),
+                    Statement::Return(ReturnStatement {
+                        value: Some(ReturnStatementValue::Expr(Expr::Identifier(Ident {
+                            value: "@foo".into(),
+                            quote_style: None,
+                            span: Span::empty(),
+                        }))),
+                    }),
+                ],
+                end_token: AttachedToken(TokenWithSpan::wrap(sqlparser::tokenizer::Token::Word(
+                    sqlparser::tokenizer::Word {
+                        value: "END".to_string(),
+                        quote_style: None,
+                        keyword: Keyword::END
+                    }
+                ))),
+            })),
+            behavior: None,
+            called_on_null: None,
+            parallel: None,
+            using: None,
+            language: None,
+            determinism_specifier: None,
+            options: None,
+            remote_connection: None,
+        }),
+    );
+
+    let create_function_with_conditional = r#"
+        CREATE FUNCTION some_scalar_udf()
+        RETURNS INT
+        AS
+        BEGIN
+            IF 1=2
+            BEGIN
+                RETURN 1;
+            END
+
+            RETURN 0;
+        END
+    "#;
+    let create_stmt = ms().one_statement_parses_to(create_function_with_conditional, "");
```

Comment:
can we use `verified_stmt` here instead (what does the empty canonical string imply)?

Reply:
I think the main downside is we lose the newlines, right? So we can't write SQL text the conventional way with new lines with `r#"..."#;`. Unless, are you suggesting this example case be reduced to _only_ verified_stmt, and no ast assertion?